projects/components/src/snackbar/snackbar.component.ts

```diff
@@ -0,0 +1,67 @@
+import { ChangeDetectionStrategy, Component, Inject } from '@angular/core';
+import { MAT_SNACK_BAR_DATA } from '@angular/material/snack-bar';
+import { IconType } from '@hypertrace/assets-library';
+import { assertUnreachable } from '@hypertrace/common';
+import { Observer } from 'rxjs';
+import { IconSize } from '../icon/icon-size';
+
+@Component({
+  selector: 'ht-snackbar',
+  styleUrls: ['./snackbar.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="snackbar-container">
+      <ht-icon
+        [ngClass]="this.matData.mode"
+        class="status-icon"
+        [icon]="this.iconType"
+        size="${IconSize.Small}"
+      ></ht-icon>
+      <div class="text">{{ this.matData.message }}</div>
+      <ht-icon
+        class="dismiss-icon"
+        icon="${IconType.CloseCircleFilled}"
+        size="${IconSize.Small}"
+        (click)="this.close()"
+      ></ht-icon>
+    </div>
+  `
+})
+export class SnackbarComponent {
+  public iconType?: IconType;
+  public constructor(@Inject(MAT_SNACK_BAR_DATA) public readonly matData: SnackbarData) {
+    this.setIconType(this.matData.mode);
+  }
+
+  public setIconType(mode: SnackbarMode): void {
+    switch (mode) {
+      case SnackbarMode.Success:
+        this.iconType = IconType.Checkmark;
+        break;
+      case SnackbarMode.Failure:
+        this.iconType = IconType.Close;
+        break;
+      case SnackbarMode.Info:
+        this.iconType = IconType.Info;
+        break;
+      default:
+        assertUnreachable(mode);
+    }
+  }
+
+  public close(): void {
+    this.matData.closedObserver.next(undefined);
```

Comment:
If we used an `Observer<void>` instead of an `Observer<unknown>` we should be able to avoid the unnecessary undefined arg.